src/main/java/org/mule/tools/apikit/model/MuleConfigBuilder.java

```diff
@@ -69,6 +69,12 @@ public static Optional<ApikitRouter> getRouter(Element flow) {
         Element flowContentElement = (Element) flowContent;
         if (elementIsApikitRouter(flowContentElement)) {
           return Optional.of(new ApikitRouter(flowContentElement));
+        } else {
+          // Recursively trying to find an element (or one of its underlying elements) that is apikit router
+          Optional<ApikitRouter> apikitRouterOptional = getRouter(flowContentElement);
```

Comment:
if we return that line directly, it will only traverse the first element of the child content list (left tree)
We want to only return if we actually found apikit router in that branch.

Reply:
Okay. I get it. You want to avoid returning a negative prematurely. But if the `flow` does not contain any router, do we return a `Optional.empty` anywhere? Do we have  a test for this?

Reply:
Added a test where no router found in the flows